README.md

```diff
@@ -1,8 +1,29 @@
 # Task
 
-Welcome to your new gem! In this directory, you'll find the files you need to be able to package up your Ruby library into a gem. Put your Ruby code in the file `lib/task`. To experiment with that code, run `bin/console` for an interactive prompt.
+Task provides a toolbox for generating, tracking and serializing tasks to be performed.
+It is NOT a job queuing framework, a la resque or qless, and does not provide ways to execute tasks
+in a distributed fashion. Rather, Task could be used with such a framework to provide "at-least-once"
+task execution guarantees, or to perform multiple, lightweight tasks within a single job.
 
-TODO: Delete this and the text above, and describe your gem
+An example task lifecycle could be follows:
+
+1. A task is generated
+2. The task is saved to a backing data store, recording that it has been generated and should be completed.
+3. The task is serialized (as JSON) and passed to another process or host to complete.
```

Comment:
interesting. i would think that the task would be serialized as JSON before being saved to the datastore.  and then upon being passed to another process or host, before a worker process could begin working on the task, it would need to be deserialized.  am i misunderstanding something?